public/v4/apps/teslamate.yml

```diff
@@ -0,0 +1,105 @@
+        volumes:
+            - $$cap_appname-db:/var/lib/postgresql/data
+    $$cap_appname-grafana:
+        caproverExtra:
+            containerHttpPort: 3000
+            websocketSupport: 'true'
+        image: teslamate/grafana:$$cap_GRAFANA_VERSION
+        restart: always
+        environment:
+            DATABASE_USER: $$cap_DATABASE_USER
+            DATABASE_PASS: $$cap_DATABASE_PASS
+            DATABASE_NAME: $$cap_DATABASE_NAME
+            DATABASE_HOST: srv-captain--$$cap_appname-db
+        volumes:
+            - $$cap_appname-db:/var/lib/postgresql/data
+        depends_on:
+            - $$cap_appname-db
+    $$cap_appname-mqtt:
+        caproverExtra:
+            notExposeAsWebApp: 'true'
+        image: eclipse-mosquitto:2
+        restart: always
+        command: mosquitto -c /mosquitto-no-auth.conf
+        volumes:
+            - $$cap_appname-mosquitto-conf:/mosquitto/config
+            - $$cap_appname-mosquitto-data:/mosquitto/data
+    $$cap_appname:
+        caproverExtra:
+            containerHttpPort: 4000
+            websocketSupport: 'true'
+        image: teslamate/teslamate:$$cap_TESLAMATE_VERSION
+        environment:
+            ENCRYPTION_KEY: $$cap_SECRET_KEY
+            DATABASE_USER: $$cap_DATABASE_USER
+            DATABASE_PASS: $$cap_DATABASE_PASS
+            DATABASE_NAME: $$cap_DATABASE_NAME
+            DATABASE_HOST: srv-captain--$$cap_appname-db
+            MQTT_HOST: srv-captain--$$cap_appname-mqtt
+        volumes:
+            - $$cap_appname-import:/opt/app/import
+        depends_on:
+            - $$cap_appname-db
+caproverOneClickApp:
+    displayName: TeslaMate
+    description: A powerful, self-hosted data logger for your Tesla.
+    isOfficial: true
+    instructions:
+        start: |-
+            TeslaMate is a powerful self-hosted data logger for your Tesla.
+        end: |-
+            TeslaMate has been successfully deployed! It might take few moments before it's fully started.
+            You can access it at `http://$$cap_appname.$$cap_root_domain` and connect your Tesla account.
+            Visit `http://$$cap_appname-grafana.$$cap_root_domain` to access Grafana dashboard. (You can log in with the default login/password of admin/admin)
+
+            Read more at `https://docs.teslamate.org/docs/installation/docker#usage` for further instructions and usage.
+
+            **Important:** Please enable **HTTPS** and **WebSocket Support** before use.
+            **Recommended:** Protect your `TeslaMate` app with **HTTP Basic Auth** in CapRover to prevent public access.
+    variables:
+        - id: $$cap_GRAFANA_VERSION
+          label: Grafana | Version
+          defaultValue: latest
+          description: Check out their Docker page for the valid tags https://hub.docker.com/r/teslamate/grafana/tags
+          validRegex: /^([^\s^\/])+$/
+        - id: $$cap_TESLAMATE_VERSION
+          label: Application | Version
+          defaultValue: latest
```

Comment:
Avoid using `latest` tag as it's unstable and it will cause the new installations to break if the new version has a breaking change.

Reply:
Changed to fixed number `1.28` from `latest`.